worker/processors/system/index.js

```diff
@@ -0,0 +1,16 @@
+const donorCheck = require('./donorCheck');
+
+module.exports = async (job) => {
+  sails.log.debug('[Worker] Got a `system` job', job.data);
+
+  switch (job.data.type) {
+    case 'donorCheck':
+      await donorCheck(job);
```

Comment:
you probably don't need to await here, nothing is waiting for the result.

Reply:
awaiting is better for error handling though, Bull catches it properly and reports the job as failed if needed

Reply:
Except you not doing that. Is the parent function awaiting? If so you should just return the promise instead of awaiting the promise.

Reply:
Right now it's not clear in that function